<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>admin</class>
 <widget class="QMainWindow" name="admin">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>996</width>
    <height>762</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="4">
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="minimumSize">
         <size>
          <width>300</width>
          <height>30</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>仿宋</family>
          <pointsize>18</pointsize>
         </font>
        </property>
        <property name="text">
         <string>欢迎登录成绩管理系统</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <spacer name="verticalSpacer_3">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Maximum</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>42</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="0" colspan="4">
     <layout class="QHBoxLayout" name="horizontalLayout_7">
      <item>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="QnameLabel">
          <property name="text">
           <string>管理员：</string>
          </property>
          <property name="buddy">
           <cstring>userName</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="userName">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>156</width>
            <height>24</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="majorLabel">
          <property name="text">
           <string>工号：</string>
          </property>
          <property name="buddy">
           <cstring>userId</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="userId">
          <property name="minimumSize">
           <size>
            <width>156</width>
            <height>24</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="3" column="0" rowspan="2">
     <spacer name="verticalSpacer_6">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Minimum</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>120</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="3" column="2">
     <spacer name="horizontalSpacer_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeType">
       <enum>QSizePolicy::Fixed</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>13</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="3" column="3">
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QComboBox" name="searchChange">
        <property name="minimumSize">
         <size>
          <width>105</width>
          <height>21</height>
         </size>
        </property>
        <item>
         <property name="text">
          <string>按学号</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>按专业班级</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_8">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="4" column="3" rowspan="2">
     <widget class="QTableWidget" name="gradeListTable">
      <property name="minimumSize">
       <size>
        <width>729</width>
        <height>549</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="3">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="modifyPassword">
              <property name="minimumSize">
               <size>
                <width>93</width>
                <height>28</height>
               </size>
              </property>
              <property name="text">
               <string>修改密码</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_9">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Maximum</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="resetStudentPW">
              <property name="minimumSize">
               <size>
                <width>101</width>
                <height>28</height>
               </size>
              </property>
              <property name="text">
               <string>重置学生密码</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="addGrade">
              <property name="text">
               <string>添加成绩</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_10">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Maximum</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="setTip">
              <property name="text">
               <string>发布通知</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QPushButton" name="update">
              <property name="minimumSize">
               <size>
                <width>93</width>
                <height>28</height>
               </size>
              </property>
              <property name="text">
               <string>更新</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_11">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Maximum</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="save">
              <property name="minimumSize">
               <size>
                <width>93</width>
                <height>28</height>
               </size>
              </property>
              <property name="text">
               <string>导出</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QPushButton" name="search">
            <property name="text">
             <string>查询</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_7">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="expel">
            <property name="text">
             <string>删除</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QPushButton" name="courseManage">
            <property name="text">
             <string>课程管理</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="quit">
            <property name="text">
             <string>退出</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_7">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>80</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>searchChange</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>admin</receiver>
   <slot>BserachChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>669</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>712</x>
     <y>167</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modifyPassword</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>BmodifyPW()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>92</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>83</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetStudentPW</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>BResetStudentPW()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setTip</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>BsetTip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>372</y>
    </hint>
    <hint type="destinationlabel">
     <x>181</x>
     <y>495</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addGrade</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>BAddGrade()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>372</y>
    </hint>
    <hint type="destinationlabel">
     <x>68</x>
     <y>484</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>update</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>Bupdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>456</y>
    </hint>
    <hint type="destinationlabel">
     <x>71</x>
     <y>588</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>Bsave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>244</x>
     <y>456</y>
    </hint>
    <hint type="destinationlabel">
     <x>188</x>
     <y>598</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>quit</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>Bquit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>633</y>
    </hint>
    <hint type="destinationlabel">
     <x>195</x>
     <y>543</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gradeListTable</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>admin</receiver>
   <slot>cellchanged(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>courseManage</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>BcourseManage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>625</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>651</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>search</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>Bsearch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>45</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>48</x>
     <y>558</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>expel</sender>
   <signal>clicked()</signal>
   <receiver>admin</receiver>
   <slot>expelGrade()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>223</x>
     <y>523</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>505</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>BserachChange(int)</slot>
  <slot>BmodifyPW()</slot>
  <slot>BResetStudentPW()</slot>
  <slot>BAddGrade()</slot>
  <slot>BsetTip()</slot>
  <slot>Bupdate()</slot>
  <slot>Bsave()</slot>
  <slot>Bquit()</slot>
  <slot>cellchanged(int,int)</slot>
  <slot>BcourseManage()</slot>
  <slot>expelGrade()</slot>
  <slot>Bsearch()</slot>
 </slots>
</ui>
